### PR TITLE
Create patient repository and remove patient functionality from name …

### LINKED ISSUE #1959: Split NameRepository into NameRepository + PatientRepository
The name repository shouldn't have functionality to search by enrolment ID, e.g. Having a single NameRepository would require extra programs permission checks for the NameRepository when using it in places unrelated programs...

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes #1959 

# 👩🏻‍💻 What does this PR do? 
- Creates a PatientRepository and uses it where appropriated.
- Removes patient specific filters and sorts from the NameRepository

Question: I copied name_store_join logic from NameRepository, i.e. patient needs to have a `customer` `name_store_join`. This is still correct, right?

### CODE REVIEW

**Comment by andreievg** [comment]
> Question: I copied name_store_join logic from NameRepository, i.e. patient needs to have a customer name_store_join. This is still correct, right?

I don't think we need to check for is_customer on name store for patient. It's only really for stores and normal names, this property in actually only on name itself in mSupply (i.e. is_customer and is_supplier is not on name_store_join in mSuppy), it was added in omSupply because we had requests to configure name to store relation in a more granular way

File: server/repository/src/db_diesel/name.rs
Comment (andreievg) [inline server/repository/src/db_diesel/name.rs:1]: Should we also filter by type (for name not patient type for patient only patient type) ? 

https://github.com/openmsupply/open-msupply/blob/79c8572aff0ba7d2c06316e6d423611091fedcd0/server/repository/src/db_diesel/name.rs#L264-L268

**Review by andreievg — COMMENTED** [review]
As per our conversation. Since we only consider patient visibility on site rather then per store, and central sync system will only sync patient that is visible on site, we don't really need to check name_store_join link to name for patient. Also means we don't need patient_belongs_to_store check. Is that right ?

**Comment by andreievg** [comment]
https://github.com/openmsupply/open-msupply/pull/1965#pullrequestreview-1525496156

File: server/repository/src/db_diesel/name.rs
Comment (clemens-msupply) [reply server/repository/src/db_diesel/name.rs:1]: you mean having a type filter in NameRepository? there is one already. For PatientRepository it always filters by patient type. Not sure if that was your question?

Comment (andreievg) [reply server/repository/src/db_diesel/name.rs:1]: Hmm, I was thinking that NameRepository would be for all non patient names and PatientRepository is only for patient names, and I couldn't see any static filters for that ?

**Comment by clemens-msupply** [comment]
> > Question: I copied name_store_join logic from NameRepository, i.e. patient needs to have a customer name_store_join. This is still correct, right?
> 
> I don't think we need to check for is_customer on name store for patient. It's only really for stores and normal names, this property in actually only on name itself in mSupply (i.e. is_customer and is_supplier is not on name_store_join in mSuppy), it was added in omSupply because we had requests to configure name to store relation in a more granular way

I removed the name_store_join check when querying patients. When creating a new patient I still add a name_store_join though to associate the patient with the current site. I use the current store_id for this, but it could really be any store_id on the site, right?

**Comment by andreievg** [comment]
> but it could really be any store_id on the site, right?

Yep, as far as I know (probably best to use the store from which patient was added)

File: server/repository/src/db_diesel/name.rs
Comment (andreievg) [reply server/repository/src/db_diesel/name.rs:1]: @clemens-msupply I am keen to approve this PR and get it merged, what do you think of my latest comment ?

Comment (clemens-msupply) [reply server/repository/src/db_diesel/name.rs:1]: sorry still don't understand.

There is this static filter in the Patient repo:
https://github.com/openmsupply/open-msupply/blob/dd2bf88987d9165f578409ccfffea4e109494100/server/repository/src/db_diesel/patient.rs#L224-L228

Or which filter do you mean?

Comment (andreievg) [reply server/repository/src/db_diesel/name.rs:1]: That's great, sorry I missed it. Do we need one like this for NameRepository or should that be all names, or did I miss a filter there too ?

**Review by andreievg — APPROVED** [review]
Thanks for changes.

https://github.com/openmsupply/open-msupply/pull/1966#discussion_r1261813044
> That's great, sorry I missed it. Do we need one like this for NameRepository or should that be all names, or did I miss a filter there too ?

I am going to ask about this in next stand up. I think using name for both is fine for just now (especially since Prescription PR's rely on it)

**Comment by clemens-msupply** [comment]
> Thanks for changes.
> 
> [#1966 (comment)](https://github.com/openmsupply/open-msupply/pull/1966#discussion_r1261813044)
> 
> > That's great, sorry I missed it. Do we need one like this for NameRepository or should that be all names, or did I miss a filter there too ?
> 
> I am going to ask about this in next stand up. I think using name for both is fine for just now (especially since Prescription PR's rely on it)

Thanks, yes have to check with @roxy-dao for prescriptions.

**Pull request merged**